services/management/grpc/actions_server.go

```diff
@@ -0,0 +1,264 @@
+
+package grpc
+
+import (
+	"context"
+
+	"github.com/percona/pmm/api/managementpb"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/services/agents"
+)
+
+type actionsServer struct {
+	r  *agents.Registry
+	db *reform.DB
+}
+
+// NewActionsServer creates Management Actions Server.
+func NewActionsServer(r *agents.Registry, db *reform.DB) managementpb.ActionsServer {
+	return &actionsServer{r, db}
+}
+
+// GetAction gets an action result.
+//nolint:lll
+func (s *actionsServer) GetAction(ctx context.Context, req *managementpb.GetActionRequest) (*managementpb.GetActionResponse, error) {
+	res, err := models.FindActionResultByID(s.db.Querier, req.ActionId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &managementpb.GetActionResponse{
+		ActionId:   res.ID,
+		PmmAgentId: res.PmmAgentID,
+		Done:       res.Done,
+		Error:      res.Error,
+		Output:     res.Output,
+	}, nil
+}
+
+// StartPTSummaryAction starts pt-summary action.
+//nolint:lll,dupl
+func (s *actionsServer) StartPTSummaryAction(ctx context.Context, req *managementpb.StartPTSummaryActionRequest) (*managementpb.StartPTSummaryActionResponse, error) {
+	res, err := models.CreateActionResult(s.db.Querier, req.PmmAgentId)
+	if err != nil {
+		return nil, err
+	}
+
+	ag, err := models.FindPMMAgentsForNode(s.db.Querier, req.NodeId)
+	if err != nil {
+		return nil, err
+	}
+
+	req.PmmAgentId, err = models.FindPmmAgentIDToRunAction(req.PmmAgentId, ag)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.r.StartPTSummaryAction(ctx, res.ID, req.PmmAgentId, []string{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &managementpb.StartPTSummaryActionResponse{
+		PmmAgentId: req.PmmAgentId,
+		ActionId:   res.ID,
+	}, nil
+}
+
+// StartPTMySQLSummaryAction starts pt-mysql-summary action.
+//nolint:lll,dupl
+func (s *actionsServer) StartPTMySQLSummaryAction(ctx context.Context, req *managementpb.StartPTMySQLSummaryActionRequest) (*managementpb.StartPTMySQLSummaryActionResponse, error) {
+	res, err := models.CreateActionResult(s.db.Querier, req.PmmAgentId)
+	if err != nil {
+		return nil, err
+	}
+
+	ag, err := models.FindPMMAgentsForService(s.db.Querier, req.ServiceId)
+	if err != nil {
+		return nil, err
+	}
+
+	req.PmmAgentId, err = models.FindPmmAgentIDToRunAction(req.PmmAgentId, ag)
+	if err != nil {
+		return nil, err
+	}
+
+	err = s.r.StartPTMySQLSummaryAction(ctx, res.ID, req.PmmAgentId, []string{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &managementpb.StartPTMySQLSummaryActionResponse{
+		PmmAgentId: req.PmmAgentId,
+		ActionId:   res.ID,
+	}, nil
+}
+
+// StartMySQLExplainAction starts mysql-explain action.
+//nolint:lll,dupl
+func (s *actionsServer) StartMySQLExplainAction(ctx context.Context, req *managementpb.StartMySQLExplainActionRequest) (*managementpb.StartMySQLExplainActionResponse, error) {
+	res, err := models.CreateActionResult(s.db.Querier, req.PmmAgentId)
```

Comment:
QAN App(Frontend) doesn't have info about pmmAgentID(at least without extra request) so it can't send it to pmm-managed. And this line will fail because of pmm_agent_id constraint.